### ISSUE TITLE
[bug] Cant set env variable CONAN_ERROR_ON_OVERRIDE in a profile

### ISSUE THREAD

**Issue by MichaelHCline** [conan-io/conan#7495]
I can set CONAN_ERROR_ON_OVERRIDE=1 in my conan.conf, or I can set the env variable on the command line:
export CONAN_ERROR_ON_OVERRIDE=1
However, putting this in my profile file, does not seem to have any affect.
[env]
CONAN_ERROR_ON_OVERRIDE=1

### Environment Details (include every applicable attribute)
  * Operating System+version:  Linux   3.10.0-1127.13.1.el7.x86_64
  * Compiler+version:  n/a
  * Conan version:  1.24
  * Python version:

**Comment by memsharded** [comment]
This is expected, at the moment the env-vars defined in the profiles apply to recipes, those env-vars will be activated in the recipes ``build()``, ``package()``, etc. methods. As described in the docs, the settings and env-vars defined in the profiles apply per-package, so

```
[settings]
os=Windows
# is equivalent to
*:os=Windows

[env]
MYVAR=1
#is equivalent to
*:MYVAR=1
```

As they are a package thing, they do not apply to core Conan dependency resolution algorithms where many packages are involved.

It is very likely that global env-vars will be removed as a mechanism for global configuration in Conan 2.0, and only conan.conf variables will be allowed. For example, this CONAN_ERROR_ON_OVERRIDE is more like a team/company policy than a profile thing. I would recommend to manage that in the conan.conf, and if necessary, manage the conan.conf with ``conan config install``.

Probably it makes sense to re-consider how env-vars are applied to recipes, and also we need to move some env-vars from conan.conf to profiles, but this will take time, and as it might be potentially breaking, we need to be careful about it, most likely something we can consider for Conan 2.0.

**Comment by memsharded** [comment]
This variable is outdated, Conan 2.0 by default errors on conflicts and overrides must be explicit. Closing as solved in 2.0
